util/dvsim/dvsim.py

```diff
@@ -207,6 +207,9 @@ def copy_repo(src, dest, dry_run):
         # TODO: hack - include hw/foundry since it is excluded in .gitignore.
         rsync_cmd += "--include=hw/foundry "
         rsync_cmd += "--exclude-from={} ".format(ignore_patterns_file)
+
+    if os.path.exists(os.path.join(src, ".*")):
```

Comment:
I don't think this is right: it checks for the existence of a file called `*`. The problem is that we end up shell expanding `rsync_cmd` twice (and then you get a glob expansion the second time). The correct fix is probably to quote `rsync_cmd` with shlex.quote.

Reply:
ah.. you are right. it should check anyfile start with `.` at root. So this basically removes `--exclude` option always... Let me check the shlex.quote example. Thanks for letting me know.

Reply:
So, as suggested, I revised the code to just have `shlex.quote()` nothing else.

Reply:
May I suggest you instead do something like this? It's probably more robust against later problems with special characters and it gets rid of all the careful "spaces in strings", which are a pain to get right.
```python
    rsync_cmd = ["rsync",
                 "--recursive", "--links",
                 "--checksum", "--update", "--inplace"]

    # Supply `.gitignore` from the src area to skip temp files.
    ignore_patterns_file = os.path.join(src, ".gitignore")
    if os.path.exists(ignore_patterns_file):
        # TODO: hack - include hw/foundry since it is excluded in .gitignore.
        rsync_cmd += ["--include=hw/foundry",
                      "--exclude-from={}".format(ignore_patterns_file),
                      "--exclude=.*"]

    rsync_cmd += [src + "/.", dest]
    rsync_str = ' '.join(shlex.quote(w for w in rsync_cmd))
    cmd = ["flock", "--timeout", "600", dest, "--command", rsync_str]
```
(I've not actually tested this, so you'll want to make sure it works, but I've used similar code in the past)

Reply:
Hum.. this does not work. shlex complains about TypeError. I will leave this task to @sriyerg .

Thanks for suggestion.

Reply:
Uh. In case anyone comes back to this, there was a typo in the last line of my suggestion. Should have been:
```
    rsync_str = ' '.join(shlex.quote(w) for w in rsync_cmd)
```